tests/csrf_tests/tests.py

```diff
@@ -91,59 +90,71 @@ def test_get_token_csrf_cookie_not_set(self):
+        self.assertNotEqual(cookie, TEST_SECRET)
         self.assertIs(request.META['CSRF_COOKIE_NEEDS_UPDATE'], True)
 
-    def test_sanitize_token_masked(self):
-        # Tokens of length CSRF_TOKEN_LENGTH are preserved.
+    def test_check_token_format_valid(self):
         cases = [
-            (MASKED_TEST_SECRET1, MASKED_TEST_SECRET1),
-            (64 * 'a', 64 * 'a'),
+            # Tokens of length CSRF_SECRET_LENGTH
+            TEST_SECRET,
+            # Tokens of length CSRF_TOKEN_LENGTH
+            MASKED_TEST_SECRET1,
+            64 * 'a',
         ]
-        for token, expected in cases:
+        for token in cases:
             with self.subTest(token=token):
-                actual = _sanitize_token(token)
-                self.assertEqual(actual, expected)
-
-    def test_sanitize_token_unmasked(self):
-        # A token of length CSRF_SECRET_LENGTH is masked.
-        actual = _sanitize_token(TEST_SECRET)
-        self.assertMaskedSecretCorrect(actual, TEST_SECRET)
+                actual = _check_token_format(token)
+                self.assertIsNone(actual)
 
-    def test_sanitize_token_invalid(self):
+    def test_check_token_format_invalid(self):
         cases = [
             (64 * '*', 'has invalid characters'),
             (16 * 'a', 'has incorrect length'),
         ]
         for token, expected_message in cases:
             with self.subTest(token=token):
                 with self.assertRaisesMessage(InvalidTokenFormat, expected_message):
-                    _sanitize_token(token)
+                    _check_token_format(token)
 
     def test_does_token_match(self):
         cases = [
-            ((MASKED_TEST_SECRET1, MASKED_TEST_SECRET2), True),
-            ((MASKED_TEST_SECRET1, 64 * 'a'), False),
+            # A masked token works.
+            ((MASKED_TEST_SECRET1, TEST_SECRET), True),
+            ((MASKED_TEST_SECRET2, TEST_SECRET), True),
+            ((64 * 'a', _unmask_cipher_token(64 * 'a')), True),
+            # An unmasked token works.
+            ((TEST_SECRET, TEST_SECRET), True),
+            ((32 * 'a', 32 * 'a'), True),
+            # Incorrect tokens don't work.
+            ((32 * 'a', TEST_SECRET), False),
+            ((64 * 'a', TEST_SECRET), False),
         ]
-        for (token1, token2), expected in cases:
-            with self.subTest(token1=token1, token2=token2):
-                actual = _does_token_match(token1, token2)
+        for (token, secret), expected in cases:
+            with self.subTest(token=token, secret=secret):
+                actual = _does_token_match(token, secret)
                 self.assertIs(actual, expected)
 
+    def test_does_token_match_wrong_token_length(self):
+        cases = [
+            16 * 'a',
+            48 * 'a',
+        ]
+        for token in cases:
+            with self.subTest(token=token):
+                with self.assertRaises(AssertionError):
+                    _does_token_match(token, TEST_SECRET)
+
```

Comment:
This assertion is unreachable in the `CsrfViewMiddleware` because tokens are validated earlier. So in this case, the `assert` merely helps when refactoring, etc. (see ticket-32508 and [Tim's comment](https://code.djangoproject.com/ticket/32508#comment:3)). IMO it's fine.